extensions/ql-vscode/src/local-databases.ts

```diff
@@ -643,6 +645,7 @@ export class DatabaseManager extends DisposableObject {
   public async addExistingDatabaseItem(
     databaseItem: DatabaseItem,
     progress: ProgressCallback,
+    makeSelected = true,
```

Comment:
I think either true or false is fine as the default value for `makeSelected`, so long as we're happy we've audited the existing call sites. But that's not quite what I was getting at.

At the point where someone calls `addExistingDatabaseItem` they don't know if the item is already in the `this._databaseItems` list or not. But if you call this and set `makeSelected = true` it'll only actually be set as selected if it's not already in the list of known database items. If it does happen to exist then we won't do anything. I find that behaviour confusing. Does that make sense?

The name being "existing" is confusing but that's a different matter. I assume it means it's being passed a `DatabaseItem` instead of making one from the `Uri`, but it's confusing because it could also mean it exists in the list of database items. But I would change that in this PR to avoid this getting out of control. Maybe raise as a follow up tech-debt issue?

Reply:
Done.